### ISSUE TITLE
WeatherNet model and training hyperparameters

### ISSUE THREAD

**Issue by ptoews** [rheinzler/PointCloudDeNoising#11]
Hi @rheinzler,

for my master's thesis I would like to compare WeatherNet with other approaches and therefore want to implement it as closely to your model as possible. However, I couldn't find information about all the hyperparameters I have to set, and wondered if you could help me with the following questions:

- What kind of padding did you use to keep the spatial dimensions in the modified LiLaBlock? I assume you added padding to compensate for the non-quadratic kernels, but to keep the spatial dimensions one has to either add padding to the four parallel conv layers or the single succeeding one.
- Did you use bias in the conv layers?
- What p did you choose for the dropout layer?
- What loss did you choose, cross-entropy loss?
- Did you weight the point classes for loss computation, as they are imbalanced? How did you choose the weights?
- Did you weight the samples (frames), as they are imbalanced as well?
- Lastly, about the learning rate. Did you really use 4e-8? Because it takes a few epochs to reduce the loss by just 0.1% in my experiments. I found 4e-3 to work the best in my case, but maybe it depends on the choice of parameters above.

Thanks in advance!

**Comment by rheinzler** [comment]
Hi @ptoews,

below you can find my answers. I hope this helps.

Q1 Padding:
- padding= same
- strides=(1, 1)

Q2 Conv Layers:
- Here you can see the tensorflow implementation for the 3x3 convolution in the WeatherBlock. 
```python
import tensorflow.contrib.slim as slim
conv_square = slim.conv2d(inputs=input_element,
                          num_outputs=num_output,
                          kernel_size=[3, 3],
                          activation_fn=tf.nn.relu,
                          padding="same")
```
Q3 What p did you choose for the dropout layer:
- WeatherNet: dropout_ratio=0.5 (for the dropout layer); dropout_ratio=0.075 in each sub-element.

Q4 Loss:
- Yes, cross-entropy loss.

Q5 Weights (for loss) due to imbalanced data:
- No, we did not applied weights, but it might be a worth a try.

Q6 Weights for samples:
- we balanced the data set by adding augmented samples for fog and rain. As a result the class distribution was balanced on a frame basis.

Q7 Learning rate:
- LiLaNet: 4e-4
- RangeNet: 1e-4
- WeatherNet: 8e-4

Sorry for the late reply. If you have any further questions, please feel free to contact me.
